common/test/acceptance/tests/lms/test_progress_page.py

```diff
@@ -0,0 +1,277 @@
+            self._check_progress_page_with_scored_problem()
+
+        edit(self)
+
+        with self._logged_in_session():
+            self.assertEqual(self._get_problem_scores(), [(1, 1), (0, 1)])
+            self.assertEqual(self._get_section_score(), (1, 2))
+
+    def test_visibility_change_does_affect_score(self):
+        with self._logged_in_session():
+            self._check_progress_page_with_scored_problem()
+
+        self._set_staff_lock_on_subsection(True)
+
+        with self._logged_in_session():
+            self.assertEqual(self._get_problem_scores(), None)
+            self.assertEqual(self._get_section_score(), None)
+
+        self._set_staff_lock_on_subsection(False)
+
+        with self._logged_in_session():
+            self.assertEqual(self._get_problem_scores(), [(1, 1), (0, 1)])
+            self.assertEqual(self._get_section_score(), (1, 2))
+
+
+class SubsectionGradingPolicyTest(ProgressPageBaseTest):
+    """
+    Tests changing a subsection's 'graded' field
+    and the effect it has on the progress page.
+    """
+    def setUp(self):
+        super(SubsectionGradingPolicyTest, self).setUp()
+        self._set_policy_for_subsection("Homework")
+
+    def _set_policy_for_subsection(self, policy):
+        """
+        Set the grading policy for the
+        subsection in the test.
+        """
+        with self._logged_in_session(staff=True):
+            self.course_outline.visit()
+            modal = self.course_outline.section_at(0).subsection_at(0).edit()
+            modal.policy = policy
+            modal.save()
+
+    def test_subsection_grading_policy_on_progress_page(self):
+        with self._logged_in_session():
+            self.assertEqual(self._get_problem_scores(), [(0, 1), (0, 1)])
+            self.assertEqual(self._get_section_score(), (0, 2))
+            self.assertTrue(self.progress_page.text_on_page("Homework 1 - Test Subsection 1 - 0% (0/2)"))
+            self.courseware_page.visit()
+            self._answer_problem_correctly()
+            self.assertEqual(self._get_problem_scores(), [(1, 1), (0, 1)])
+            self.assertEqual(self._get_section_score(), (1, 2))
+            self.assertTrue(self.progress_page.text_on_page("Homework 1 - Test Subsection 1 - 50% (1/2)"))
+
+        self._set_policy_for_subsection("Not Graded")
+
+        with self._logged_in_session():
+            self.progress_page.visit()
+            self.assertEqual(self._get_problem_scores(), [(1, 1), (0, 1)])
+            self.assertEqual(self._get_section_score(), (1, 2))
+            self.assertFalse(self.progress_page.text_on_page("Homework 1 - Test Subsection 1"))
+
+        self._set_policy_for_subsection("Homework")
+        with self._logged_in_session():
+            self.assertEqual(self._get_problem_scores(), [(1, 1), (0, 1)])
+            self.assertEqual(self._get_section_score(), (1, 2))
+            self.assertTrue(self.progress_page.text_on_page("Homework 1 - Test Subsection 1 - 50% (1/2)"))
```

Comment:
Nit: consider writing an internal helper method for the above 3 statements, so it can be reused 4 times within this method.